application-ideas-ui/src/main/resources/Ideas/Migrations/1.14/WebHome.xml

```diff
@@ -0,0 +1,116 @@
+
+#macro (handleWiki $w)
+  #set ($ideasClass = 'Ideas.IdeasClass')
+  #set ($ideasClassDoc = $xwiki.getDocument("${w}:$ideasClass"))
+  #set ($wikiPrettyName = $services.wiki.getById($w).prettyName)
+  #if ("$!wikiPrettyName.trim()" == '')
+    #set ($wikiPrettyName = $w)
+  #end
+  #if ($ideasClassDoc)
+    #set ($ideas = $services.query.xwql("from doc.object($ideasClass) as ideas order by doc.fullName asc").setWiki($w).addFilter('unique').execute())
+    $services.localization.render('ideas.migrations.wiki') $wikiPrettyName
+    $services.localization.render('ideas.migrations.number') $ideas.size()
+
+    ## Table header
+    |=$services.localization.render('ideas.migrations.document')##
+    |=$services.localization.render('ideas.migrations.status')##
+    |=$services.localization.render('ideas.migrations.properties')
+    #foreach ($md in $ideas)
+      #set ($mDocRef = $services.model.resolveDocument($md, $services.model.createWikiReference($w)))
+      #set ($mDoc = $xwiki.getDocument($mDocRef))
+      #set ($ideasObj = $mDoc.getObject($ideasClass))
+      #set ($nbestimate = $ideasObj.getProperty('nbestimate').value)
+      #set ($saved = $NULL)
+      #if($nbestimate)
+        #set($stringCost = "$nbestimate")
+        ## Save the idea's cost value after converting to the string.
+        #set ($saved = 'Would be saved on actual run')
+        #if ("$!request.confirm" == 'true')
+          #set($discard = $ideasObj.set('cost', $stringCost))
+          #set ($discard = $mDoc.save('Move the idea''s estimatecost converted value to the cost'))
+          #set ($saved = 'Saved')
+        #end
+      #end
+      ## Clean the deprecated properties, if there are any
+      #set ($documentToClean = $xwiki.xWiki.getDocument($mDocRef, $xcontext.context))
+      #set ($deprecatedBaseObject = $documentToClean.getObject($ideasClass))
+      #set ($needsSave = false)
+      #foreach ($baseProp in $deprecatedBaseObject.getXClass($xcontext.context).getDeprecatedObjectProperties($deprecatedBaseObject))
+        #if ($baseProp.name == 'nbestimate')
+          #set ($cleanedProps = $baseProp.name)
+          #if ("$!request.confirm" == 'true')
+            #set ($discard = $deprecatedBaseObject.removeField($baseProp.name))
+            #set ($needsSave = true)
+         #end
+        #end
+      #end
+      #if ("$!request.confirm" == 'true' &amp;&amp; $needsSave)
+        ## Set the new author
+        #set ($discard = $documentToClean.setAuthorReference($xcontext.context.userReference))
+        #set ($discard = $xwiki.xWiki.saveDocument($documentToClean,
+          $services.localization.render('core.model.xobject.synchronizeObjects.versionSummary'), true, $xcontext.context))
+      #end
+      |[[$mDoc.prefixedFullName]]|$!saved|$cleanedProps
+    #end
+
+  #end
+#end
+
+#if ("$!request.confirm" != 'true')
```

Comment:
Can you add also a screenshot for how this page looks before and after a migration?

Reply:
Before
![image](https://user-images.githubusercontent.com/56109799/197463689-90cde9bf-9dbe-43a5-b464-4a04adb71132.png)
After
![image](https://user-images.githubusercontent.com/56109799/197463901-2ce7d1c7-32e4-426f-b550-f1981d3c348c.png)